extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/MicrometerTimedInterceptor.java

```diff
@@ -15,16 +17,18 @@
 import io.micrometer.core.instrument.MeterRegistry;
 import io.micrometer.core.instrument.Tags;
 import io.micrometer.core.instrument.Timer;
+import io.quarkus.arc.ArcInvocationContext;
 
 /**
  * Quarkus defined interceptor for types or methods annotated with {@link Timed @Timed}.
- *
- * @see Timed
  */
 @Interceptor
-@MicrometerTimed
+// For some unknown reason, the default value for Timed#value() is not always available in the Jandex index when the annotation literal is being processed by ArC
+// As a workaround we do specify the value explicitly - note that is has no effect on resolution because it's non-binding 
+@Timed("")
```

Comment:
I spent a lot of time analyzing this problem and I have to admit that I did not find the cause. Some integration tests just fail intermittently when using the micrometer extension. I saw test failures in the `main`, `grpc` and various `kafka-*` integration tests.

All I know is that sometimes the default value of `Timed#value()` is `null` (instead of an empty string) and then the `io.quarkus.arc.processor.AnnotationLiteralProcessor.process()` method fails with `Value is not set for Timed.value(). Most probably an older version of Jandex was used...`. What's really interesting is the fact that default values of other annotation members are OK (e.g. `Timed#description()` is an empty string).

If there's anyone who wants to spend a nice time with debugging:
1. Replace `@Timed("")` with `@Timed` and build quarkus,
2. Run the test suite of `quarkus-grpc-deployment` - some tests should fail intermittently.

@Ladicek and @manovotn might be interested as important Jandex and ArC contributors ;-).

Reply:
In case anyone stumbles upon this in the future, the cause is that when interning `MethodInternal` objects, Jandex doesn't pay attention to the `defaultValue` field in the `equals` and `hashCode` methods: https://github.com/wildfly/jandex/blob/2.3.0.Final/src/main/java/org/jboss/jandex/MethodInternal.java#L111-L161 This then leads to `MethodInternal` objects being shared when they shouldn't. The fix is simple, I'll submit a PR to Jandex.

Reply:
Thanks for the investigation Ladislav!!!